apps/rush-lib/src/cli/utilities/pnpm/PnpmShrinkwrapFile.ts

```diff
@@ -115,7 +115,18 @@ export class PnpmShrinkwrapFile extends BaseShrinkwrapFile {
       return undefined;
     }
 
-    return packageDescription.dependencies[dependencyName];
+    const version: string = packageDescription.dependencies[dependencyName];
+    if (version) {
+      // e.g.:  "/gulp-karma/0.0.5/karma@0.13.22"
+      // capture 0: "/gulp-karma/0.0.5/"
+      // capture 1: "0.0.5"
+      const specialVersionCase: RegExp = /\/[a-z\-]+\/(.*)\//;
```

Comment:
>[a-z\-]+ [](start = 44, length = 8)

NPM technically allows weird characters in a package name:

[https://www.npmjs.com/package/validate-npm-package-name#naming-rules](https://www.npmjs.com/package/validate-npm-package-name#naming-rules)

What is the usefulness of this validation?  Would it be more correct to test for something like:

^\/[^/]*\/([^/]+)\/[^/]*$